rust/connlib/libs/tunnel/src/iface_handler.rs

```diff
@@ -136,63 +136,65 @@ where
         src: &mut [u8],
         dst: &mut [u8],
     ) -> Result<()> {
-        if let Some(r) = self.check_for_dns(src) {
-            match r {
-                dns::SendPacket::Ipv4(r) => self.write4_device_infallible(device_writer, &r[..]),
-                dns::SendPacket::Ipv6(r) => self.write6_device_infallible(device_writer, &r[..]),
-            }
-            return Ok(());
-        }
+        // if let Some(r) = self.check_for_dns(src) {
+        //     match r {
+        //         dns::SendPacket::Ipv4(r) => device_writer.write4(&r[..])?,
+        //         dns::SendPacket::Ipv6(r) => device_writer.write6(&r[..])?,
+        //     };
+        //     return Ok(());
+        // }
 
-        let dst_addr = match Tunn::dst_address(src) {
-            Some(addr) => addr,
-            None => return Err(Error::BadPacket),
-        };
+        // let dst_addr = match Tunn::dst_address(src) {
+        //     Some(addr) => addr,
+        //     None => return Err(Error::BadPacket),
+        // };
 
-        let encapsulated_packet = {
-            match self.peers_by_ip.read().longest_match(dst_addr).map(|p| p.1) {
-                Some(peer) => peer.encapsulate(src, dst)?,
-                None => {
-                    self.connection_intent(src, &dst_addr);
-                    return Ok(());
-                }
-            }
-        };
+        // let encapsulated_packet = {
+        //     match self.peers_by_ip.read().longest_match(dst_addr).map(|p| p.1) {
+        //         Some(peer) => peer.encapsulate(src, dst)?,
+        //         None => {
+        //             self.connection_intent(src, &dst_addr);
+        //             return Ok(());
+        //         }
+        //     }
+        // };
 
-        self.handle_encapsulated_packet(encapsulated_packet, &dst_addr)
-            .await
+        Ok(())
+        // self.handle_encapsulated_packet(encapsulated_packet, &dst_addr)
+        //     .await
     }
 
     #[tracing::instrument(level = "trace", skip(self, iface_config, device_io))]
-    pub(crate) async fn iface_handler(
+    async fn iface_handler(
         self: &Arc<Self>,
         iface_config: Arc<IfaceConfig>,
         device_io: DeviceIo,
-    ) {
-        let device_writer = device_io.clone();
-        loop {
-            let mut src = [0u8; MAX_UDP_SIZE];
-            let mut dst = [0u8; MAX_UDP_SIZE];
-            let res = {
-                // TODO: We should check here if what we read is a whole packet
-                // there's no docs on tun device on when a whole packet is read, is it \n or another thing?
-                // found some comments saying that a single read syscall represents a single packet but no docs on that
-                // See https://stackoverflow.com/questions/18461365/how-to-read-packet-by-packet-from-linux-tun-tap
-                match device_io.read(&mut src[..iface_config.mtu()]).await {
```

Comment:
Hmm, the new `read` isn't bound by `mtu` which we do here.

This would be `1500` bytes vs `1280` I believe. Could that be the issue?